### ISSUE TITLE
Cannot promote Testflight App to store

### ISSUE THREAD

**Issue by JKennedy24** [Microsoft/app-store-vsts-extension#75]
Initially I tried to promote testflight app and I got the following:

> !] Could not find a version to edit for app 'FieldStrike'[0m

I then created a version in the Itunes connect webpage and tried again and got the error:

> 2017-08-09T12:40:40.1344190Z [!] The request could not be completed because:
> 2017-08-09T12:40:40.1360860Z 	[en-GB]: You must detail what's new in this version in English (U.K.).[0m

**Comment by JKennedy24** [comment]
I'm guessing [this is the line](https://github.com/Microsoft/app-store-vsts-extension/blob/master/Tasks/app-store-promote/app-store-promote.ts#L141) the first error is occuring because it says `Submit_Build` but the version doesn't exist yet.

Has this ever been fully tested?

**Comment by madhurig** [comment]
@JKennedy24: I am able to submit my build for review that was previously uploaded to test flight with the promote task. 

Do you upload the app to TestFlight manually or via the tasks in this extension? How do you update your app versions?

The 2nd issue seems to be like this: https://github.com/fastlane/fastlane/issues/7016. It should not be needed to specify the "What's new" during promote if that was set on the build during upload to "TestFlight"

Thanks,
Madhuri

**Comment by JKennedy24** [comment]
@madhurig These are the steps I followed:

- I uploaded the TestFlight using the extension.
- Manually distribute to external testers
- I don't add any new versions in iTunes connect (this could be where the problem lies)
- I then try to promote my app from test flight to production
and it throws the error - "Could not find a version to edit for app 'MyApp'�[0m"

Thanks for the link for Bug 2. It looks like the same error. Will this just fix itself on next FastLane update?

**Comment by madhurig** [comment]
@JKennedy24: As per my understanding, the build number has to be updated to be able to upload new build to test flight. E.g. if I previously have 2.0.1 uploaded, I have to update build number to 2.0.2 to be able to upload new version to test flight. Otherwise we see error like "Redundant Binary Upload". Once 2.0.2 is finished processing, I can use the promote task to put that build in production from test flight by selecting "latest build" in the task. 

Can you email me the settings you have enabled in the release and promote tasks along with the logs at madhurig at microsoft dot com?

Thanks,
Madhuri

**Comment by JKennedy24** [comment]
@madhurig  you are correct. I manage to change my version number on my binary and upload to testflight successfully.

It's the promoting from testflight to production where the error occurs.

I will try this again next release cycle (which will hopefully be next week ) and if I still get the same error I will email across my settings

**Comment by JKennedy24** [comment]
@madhurig I have just tried again during our latest release cycle and still got the same error:

2017-11-16T14:46:44.7478940Z [!] Could not find a version to edit for app 'MyApp'[0m
2017-11-16T14:46:45.2207320Z ##[error]Error: fastlane failed with return code: 1

My setup of the release task is pretty much the default. the only difference I can see if that I have unticked `Release Automatically` in the "Submit to app store for review"

**Comment by madhurig** [comment]
@JKennedy24 : Can you try setting additional fastlane argument "--app-version <higherVersionCode>" and see if it will create a new version and successfully upload. 

Thanks,
Madhuri

**Comment by JKennedy24** [comment]
@madhurig Thanks for the reply. I have added this argument and will try this out next release cycle and let you know

**Comment by JKennedy24** [comment]
@madhurig  I have just tried promoting a testflight app to app store using the fastlane argument as suggested but failed with the following reason:

> started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
> 2017-12-07T09:24:04.9164650Z invalid option: --app-version higherVersionCode
> 2017-12-07T09:24:05.4170560Z ##[error]Error: fastlane failed with return code: 1

Is it possible this argument isn't valid or have I done something wrong?

**Comment by madhurig** [comment]
@JKennedy24: I meant you have to pass a higher version code that is valid to --app-version, not the literal string "higherVersionCode". Can you try that please?

Thanks,
Madhuri

**Comment by JKennedy24** [comment]
@madhurig I'll try that. That would mean I would need to set that version code every time I upload which isn't great but hopefully it will help diagnose the problem
